### PR TITLE
Azure retry hardening

### PR DESCRIPTION
**Pull request by h-dub**
Azure retry hardening
* Removed 413 as a retry-able error code
* Attempt retries even when azure fails to send a header

### CODE REVIEW

File: tools/c7n_azure/c7n_azure/utils.py [hunk 1]
```diff
@@ -140,13 +141,17 @@ def custodian_azure_send_override(self, request, headers=None, content=None, **k
...
             for k in response.headers.keys():
                 if StringUtils.equal('retry-after', k):
                     retry_after = int(response.headers[k])
-
-            if retry_after is not None and retry_after < constants.DEFAULT_MAX_RETRY_AFTER:
+            if retry_after is None:
+                # we want to attempt retries even when azure fails to send a header
+                # this has been a constant source of instability in larger environments
+                retry_after = (constants.DEFAULT_RETRY_AFTER * retries) + \
+                    random.randint(1, constants.DEFAULT_RETRY_AFTER)
```
Comment (h-dub) [inline tools/c7n_azure/c7n_azure/utils.py:153]: will add a test :)

Comment (h-dub) [reply tools/c7n_azure/c7n_azure/utils.py:153]: fixed

Comment (h-dub) [reply tools/c7n_azure/c7n_azure/utils.py:153]: verified functional passes without throttling errors with this change

**Review by kapilt — APPROVED** [review]
lgtm, thanks

**Pull request merged**